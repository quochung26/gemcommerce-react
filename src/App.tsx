import UnitValue from "./components/UnitValue";

const App = () => {
  return (
    <div className="w-screen h-screen bg-neutral-850 flex items-center justify-center text-neutral-100">
      <UnitValue />
    </div>
  );
};

export default App;
