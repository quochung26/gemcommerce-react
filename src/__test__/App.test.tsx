import { render } from "@testing-library/react";
import { expect, describe, it } from "vitest";
import App from "../App";

describe('App', () => {
  it('renders without crashing', () => {
    const { container } = render(<App />);
    expect(container).toBeDefined();
  });

  it('renders with correct background and layout classes', () => {
    const { container } = render(<App />);
    const mainDiv = container.firstChild as HTMLElement;
    expect(mainDiv.className).toContain('w-screen');
    expect(mainDiv.className).toContain('h-screen');
    expect(mainDiv.className).toContain('bg-neutral-850');
    expect(mainDiv.className).toContain('flex');
    expect(mainDiv.className).toContain('items-center');
    expect(mainDiv.className).toContain('justify-center');
  });

  it('renders UnitValue component', () => {
    const { container } = render(<App />);
    
    // Check if UnitValue component is rendered with its unique class combinations
    const unitValueContainer = container.querySelector('.bg-neutral-925.p-4.flex.flex-col');
    expect(unitValueContainer).toBeDefined();

    // Find the tab container
    const tabContainer = container.querySelector('.bg-neutral-875.rounded-lg.p-0\\.5');
    expect(tabContainer).toBeDefined();
    
    // Verify the unit options are present
    const buttons = tabContainer?.querySelectorAll('button');
    expect(buttons?.length).toBe(2);
    const buttonTexts = Array.from(buttons || []).map(btn => btn.textContent);
    expect(buttonTexts).toContain('%');
    expect(buttonTexts).toContain('px');
  });
});
