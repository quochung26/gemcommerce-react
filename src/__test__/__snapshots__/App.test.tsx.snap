// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`toUpperCase 1`] = `
<DocumentFragment>
  <div
    class="w-screen h-screen bg-neutral-850 flex items-center justify-center text-neutral-100"
  >
    <div
      class="w-70 bg-neutral-925 p-4 flex flex-col gap-4 text-xs"
    >
      <div
        class="flex gap-2 justify-between"
      >
        <p
          class="flex text-sm text-neutral-450 items-center font-normal"
        >
          Unit
        </p>
        <div
          class="w-35 bg-neutral-875 rounded-lg p-0.5 flex gap-0.5 h-9 font-medium"
        >
          <button
            aria-pressed="true"
            class="w-1/2 flex justify-center items-center rounded-md cursor-pointer focus-visible:outline-none hover:bg-neutral-750 bg-neutral-750"
          >
            %
          </button>
          <button
            aria-pressed="false"
            class="w-1/2 flex justify-center items-center rounded-md cursor-pointer focus-visible:outline-none hover:bg-neutral-750"
          >
            px
          </button>
        </div>
      </div>
      <div
        class="flex gap-2 justify-between"
      >
        <label
          class="flex text-sm text-neutral-450 items-center font-normal"
          for="step-input"
        >
          Value
        </label>
        <div
          class=" w-35 bg-neutral-875 rounded-lg flex h-9 font-medium border border-neutral-875 focus-within:border-[#3C67FF]"
        >
          <input
            class="peer order-2 w-full appearance-none text-center hover:bg-neutral-725 focus:outline-none"
            id="step-input"
            name="value"
            type="text"
            value="0"
          />
          <div
            class="order-1 relative group peer-hover:bg-neutral-725 hover:bg-neutral-725 rounded-l-lg"
          >
            <button
              class="relative w-9 p-2 flex items-center justify-center cursor-pointer rounded-l-lg"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M4 9.75C4 9.33579 4.33579 9 4.75 9L15.25 9C15.6642 9 16 9.33579 16 9.75C16 10.1642 15.6642 10.5 15.25 10.5H4.75C4.33579 10.5 4 10.1642 4 9.75Z"
                  fill="#AAAAAA"
                  fill-rule="evenodd"
                />
              </svg>
            </button>
            <div
              class="absolute left-1/2 -translate-x-1/2 bottom-full mb-3 hidden group-hover:flex"
            >
              <div
                class="relative bg-neutral-800 text-white text-xs px-1 py-2 rounded-md whitespace-nowrap"
              >
                Value must greater than 0
              </div>
              <div
                class="absolute left-1/2 -translate-x-1/2 top-full w-0 h-0
        border-l-6 border-l-transparent 
        border-r-6 border-r-transparent 
        border-t-6 border-t-neutral-800"
              />
            </div>
          </div>
          <div
            class="order-3 relative group peer-hover:bg-neutral-725 hover:bg-neutral-725 rounded-r-lg"
          >
            <button
              class=" w-9 p-2 flex items-center justify-center peer-hover:bg-neutral-725 cursor-pointer rounded-r-lg"
            >
              <svg
                fill="none"
                height="20"
                viewBox="0 0 20 20"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.75 4.75C10.75 4.33579 10.4142 4 10 4C9.58579 4 9.25 4.33579 9.25 4.75V9.25H4.75C4.33579 9.25 4 9.58579 4 10C4 10.4142 4.33579 10.75 4.75 10.75H9.25L9.25 15.25C9.25 15.6642 9.58579 16 10 16C10.4142 16 10.75 15.6642 10.75 15.25V10.75H15.25C15.6642 10.75 16 10.4142 16 10C16 9.58579 15.6642 9.25 15.25 9.25H10.75V4.75Z"
                  fill="#F9F9F9"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
